test/controllers/update/overseas.entity.presenter.controller.spec.ts

```diff
@@ -220,4 +234,98 @@ describe("OVERSEAS ENTITY PRESENTER controller", () => {
       expect(resp.text).toContain(`${FOUND_REDIRECT_TO} ${UPDATE_DO_YOU_WANT_TO_MAKE_OE_CHANGE_URL}`);
     });
   });
+
+  describe("POST tests on a remove journey", () => {
+    test(`redirect to ${UPDATE_DO_YOU_WANT_TO_MAKE_OE_CHANGE_PAGE} page after a successful post from overseas entity presenter page`, async () => {
+      const resp = await request(app).post(`${OVERSEAS_ENTITY_PRESENTER_URL}${JOURNEY_REMOVE_QUERY_PARAM}`).send(PRESENTER_OBJECT_MOCK);
+
+      expect(resp.status).toEqual(302);
+      expect(resp.text).toContain(`${FOUND_REDIRECT_TO} ${UPDATE_DO_YOU_WANT_TO_MAKE_OE_CHANGE_URL}`);
+      expect(mockSaveAndContinue).toHaveBeenCalledTimes(1);
+    });
+
+    test(`redirect to the ${UPDATE_DO_YOU_WANT_TO_MAKE_OE_CHANGE_PAGE}${JOURNEY_REMOVE_QUERY_PARAM} page after a successful post from overseas entity presenter page with special characters`, async () => {
+      const resp = await request(app).post(`${OVERSEAS_ENTITY_PRESENTER_URL}${JOURNEY_REMOVE_QUERY_PARAM}`).send(PRESENTER_WITH_SPECIAL_CHARACTERS_FIELDS_MOCK);
+
+      expect(resp.status).toEqual(302);
+      expect(resp.text).toContain(`${FOUND_REDIRECT_TO} ${UPDATE_DO_YOU_WANT_TO_MAKE_OE_CHANGE_URL}`);
+      expect(mockSaveAndContinue).toHaveBeenCalledTimes(1);
+    });
+
+    test("renders the current page with error message", async () => {
+      const resp = await request(app).post(`${OVERSEAS_ENTITY_PRESENTER_URL}${JOURNEY_REMOVE_QUERY_PARAM}`).send({ email: '' });
+
+      expect(resp.status).toEqual(200);
+      expect(resp.text).toContain(REMOVE_OVERSEAS_ENTITY_PRESENTER_PAGE_TITLE);
+      expect(resp.text).toContain(ErrorMessages.FULL_NAME);
+      expect(resp.text).toContain(ErrorMessages.EMAIL);
+      expect(resp.text).not.toContain(ErrorMessages.MAX_EMAIL_LENGTH);
+      expect(resp.text).not.toContain(ErrorMessages.EMAIL_INVALID_FORMAT);
```

Comment:
i would also check that the backlink is in the resp.text and that it is `${config.UPDATE_OVERSEAS_ENTITY_CONFIRM_URL}${config.JOURNEY_REMOVE_QUERY_PARAM}`

Reply:
added test